xs/src/libslic3r/Model.hpp

```diff
@@ -28,221 +28,529 @@ typedef std::vector<ModelObject*> ModelObjectPtrs;
+    /// \param instance the ModelInstance to be copied
+    /// \return ModelInstance* a pointer to the new instance
     ModelInstance* add_instance(const ModelInstance &instance);
+
+    /// Delete a ModelInstance.
+    /// \param idx size_t the index of the ModelInstance to be deleted
     void delete_instance(size_t idx);
+
+    /// Delete the last created ModelInstance object.
     void delete_last_instance();
+
+    /// Delete all ModelInstance objects found in the current ModelObject.
     void clear_instances();
 
+    /// Get the bounding box of the *transformed* instances.
     BoundingBoxf3 bounding_box();
+
+    /// Invalidate the bounding box in the current ModelObject.
     void invalidate_bounding_box();
 
+    /// Repair all TriangleMesh objects found in each ModelVolume.
     void repair();
+
+    /// Flatten all volumes and instances into a single mesh and applying all the ModelInstances transformations.
     TriangleMesh mesh() const;
+
+    /// Flatten all volumes and instances into a single mesh.
     TriangleMesh raw_mesh() const;
+
+    /// Get the raw bounding box
+    /// \return BoundingBoxf3
     BoundingBoxf3 raw_bounding_box() const;
+
+    /// Get the bounding box of the *transformed* given instance.
+    /// \param instance_idx size_t the index of the ModelInstance in the ModelInstance vector
+    /// \return BoundingBoxf3 the bounding box at the given index
     BoundingBoxf3 instance_bounding_box(size_t instance_idx) const;
+
+    /// Align the current ModelObject to ground by translating in the z axis the needed units.
     void align_to_ground();
+
+    /// Center the current ModelObject to origin.
     void center_around_origin();
+
+    /// Translate the current ModelObject with (x,y,z) units.
+    /// \param vector Vectorf3 the translation vector
     void translate(const Vectorf3 &vector);
+
+    /// Translate each TriangleMesh in every ModelVolume in this ModelObject.
+    /// \param x coordf_t the x units
+    /// \param y coordf_t the y units
+    /// \param z coordf_t the z units
     void translate(coordf_t x, coordf_t y, coordf_t z);
+
+    /// Scale the current ModelObject
+    /// \param factor float the scaling factor
     void scale(float factor);
+
+    /// Scale each TriangleMesh in every ModelVolume in this ModelObject.
+    /// \param versor Pointf3 the scaling factor in a 3d vector.
     void scale(const Pointf3 &versor);
+
+    /// Scale the current ModelObject to fit.
+    /// \param size Sizef3 the size vector
     void scale_to_fit(const Sizef3 &size);
+
+    /// Rotate the current ModelObject.
+    /// \param angle float the angle in radians
+    /// \param axis Axis the axis to be rotated around
     void rotate(float angle, const Axis &axis);
+
+    /// Mirror the current Model around a certain axis.
+    /// \param axis Axis enum member
     void mirror(const Axis &axis);
+
+    /// Transform the current ModelObject by a certain ModelInstance attributes.
```

Comment:
It's important to document that the inverse transformation is applied to all the object instances, so that the final size/position/rotation of the transformed objects doesn't change. This method is useful for flattening the instance transformations (of one instance) to the mesh.